gradle-plugin/it/src/test/java/org/eclipse/jkube/gradle/plugin/tests/FileSecretIT.java

```diff
@@ -47,11 +55,46 @@ public void ocResource_whenRun_generatesK8sSecret() throws IOException, ParseExc
     final BuildResult result = gradleRunner.withITProject("secret-file").withArguments("ocResource").build();
     // Then
     ResourceVerify.verifyResourceDescriptors(gradleRunner.resolveDefaultOpenShiftResourceFile(),
-        gradleRunner.resolveFile("expected", "openshift.yml"));
+        adjustSecretEncoding(gradleRunner.resolveFile("expected", "openshift.yml")));
     assertThat(result).extracting(BuildResult::getOutput).asString()
         .contains("Running in OpenShift mode")
         .contains("Using resource templates from")
         .contains("Adding revision history limit to 2")
         .contains("validating");
   }
+
+  /**
+   * On Windows, the separator used across lines is "\r\n" in contrast to *nix system where it's "\n".
+   * This function expects as input the file used for comparison in the tests and, if the test(s) is(are) running on
+   * Windows, then re-encodes the secret in them so that the Base64 string is the same between the expected file and the
+   * generated one.
+   * This is necessary because the comparison file contains the Base64 string calculated from a *nix machine.
+   *
+   * @param expectedFile the file used for comparison
+   * @return a file which has the same content as {@code expectedFile}, but with the secret re-encoded
+   */
+  private File adjustSecretEncoding(File expectedFile) throws IOException {
```

Comment:
We should probably fix this via .gitattributes so we don't have to mess with file manipulation in tests or in utility classes.

Reply:
I'll give it a try. It's a good idea.

Reply:
I checked this again and `.gitattributes` won't help since the secrets are already encoded in Base64 with the *nix line ending.
How would you use `.gitattributes` to re-encode the secret present in the file?